src/core/pool.rs

```diff
@@ -67,6 +67,54 @@ unsafe impl Allocator for Pool {
             ngx_pfree(self.0.as_ptr(), ptr.as_ptr().cast());
         }
     }
+
+    unsafe fn grow(
+        &self,
+        ptr: NonNull<u8>,
+        old_layout: Layout,
+        new_layout: Layout,
+    ) -> Result<NonNull<[u8]>, AllocError> {
+        debug_assert!(
+            new_layout.size() >= old_layout.size(),
+            "`new_layout.size()` must be greater than or equal to `old_layout.size()`"
+        );
+        self.resize_in_place(ptr, old_layout, new_layout)
+    }
+
+    unsafe fn grow_zeroed(
+        &self,
+        ptr: NonNull<u8>,
+        old_layout: Layout,
+        new_layout: Layout,
+    ) -> Result<NonNull<[u8]>, AllocError> {
+        debug_assert!(
+            new_layout.size() >= old_layout.size(),
+            "`new_layout.size()` must be greater than or equal to `old_layout.size()`"
+        );
+        self.resize_in_place(ptr, old_layout, new_layout)
+            .inspect(|new_ptr| {
+                unsafe {
+                    ptr::write_bytes(
+                        new_ptr.as_ptr().cast::<u8>().byte_add(new_layout.size()),
+                        0,
+                        old_layout.size() - new_layout.size(),
```

Comment:
if ` new_layout.size() >= old_layout.size()`

`old_layout.size() - new_layout.size()` will be negative?

Reply:
```rust
ptr::write_bytes(
    new_ptr.as_ptr().cast::<u8>().add(old_layout.size()),  // Start at end of old data
    0,
    new_layout.size() - old_layout.size(),  // Zero the NEW bytes
)
```